src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,194 @@
+  selectedOptions?: OptionProps[];
+  size?: SelectSizes;
+  onSelectedOption: Function;
+}
+
+export interface OptionsConfig extends DropdownProps {
+  indexGroup?: number;
+}
+
+const isDisabled = (
+  disabledThisOption?: boolean,
+  disabledAllOptions?: boolean
+): boolean => {
+  const result = disabledAllOptions || disabledThisOption;
+  return result || false;
+};
+const isSelectingNotMultiple = (
+  isSelecting?: boolean,
+  multiple?: boolean
+): boolean => {
+  const result = !isSelecting && !multiple;
+  return result || false;
+};
+
+const Options = ({
+  disabled = false,
+  options = [],
+  multiple = false,
+  onSelectedOption,
+  indexGroup,
+}: OptionsConfig) => {
+  const [optionsCopy, setOptionsCopy] = useState<OptionProps[]>(options);
+
+  const select = (index: number): void => {
+    if (isDisabled(optionsCopy[index].disabled, disabled)) return;
+    if (isSelectingNotMultiple(optionsCopy[index].selected, multiple)) {
+      optionsCopy.forEach((option) => {
+        option.selected = false;
+      });
+    }
+    updateSelecteds(index);
+  };
+  const updateSelecteds = (index: number): void => {
+    optionsCopy[index].selected = !optionsCopy[index].selected;
+    setOptionsCopy([...optionsCopy]);
+    onSelectedOption([...optionsCopy]);
+  };
+
+  return (
+    <ul
+      data-testid={
+        indexGroup
+          ? 'ion-dropdown-options-' + indexGroup
+          : 'ion-dropdown-options'
+      }
+    >
+      {options &&
+        options.map((option, index) => {
+          return (
+            <OptionStyle
+              key={index}
+              data-testid={
+                indexGroup
+                  ? 'ion-option-' + index + '-' + indexGroup
+                  : 'ion-option-' + index
+              }
+              disabled={disabled || option.disabled}
+              selected={option.selected}
+              onClick={() => select(index)}
+            >
+              <div className="label">{option.label}</div>
+              {option.selected && (
+                <>
+                  <div className="icon check">
+                    <IonIcon type={'check'} size={20} />
+                  </div>
+                  <div className="icon close">
+                    <IonIcon type={'close'} size={20} />
+                  </div>
+                </>
+              )}
+            </OptionStyle>
+          );
+        })}
+    </ul>
+  );
+};
+
+const isSelected = (
+  optionToCompare: OptionProps,
+  optionsUpdated: OptionProps[]
+): boolean => {
+  return optionsUpdated.some(
+    (option) => option.selected && option.value === optionToCompare.value
+  );
+};
+
+const OptionsGroup = ({
+  disabled = false,
+  optionsGroup = [],
+  onSelectedOption,
+  multiple = false,
+}: DropdownProps) => {
+  const [groupCopy, setGroupCopy] = useState<OptionGroupProps[]>(optionsGroup);
+  const handleSelect = (event: OptionProps[]) => {
+    if (!multiple) {
+      let safeCopy: OptionGroupProps[] = [...groupCopy];
```

Comment:
You tried without this copy?

Reply:
Yas, it just don't rerender da component. I know that's sounds like redundant, but was the only way that works. Using groupCopy or even optionsGroup directly, doesn't work here.